### PR TITLE
Add NetworkNamespacePath= to unit files

### PR DESCRIPTION
**Pull request by poettering**
Also, ensure that PrivateNetwork=/JoinsNamespaceOf=/NetworkNamespacePath= have an effect on socket units.

### CODE REVIEW

**Comment by boucman** [comment]
A bit of a naive question, but how would you used that ? if you need to point it to something to /proc/pid/ns, you need a pid... how can you hardcode a pid in a unit file (except for pid 1 I guess)

**Comment by poettering** [comment]
> A bit of a naive question, but how would you used that ? if you need to point it to something to /proc/pid/ns, you need a pid... how can you hardcode a pid in a unit file (except for pid 1 I guess)

You bind mount it somewhere. This is what "ip setns" does for example. It mounts them to /run/netns/<foobar>. With this option you can now use "ip setns" to set something up if you like, and then run a service inside it.

**Comment by boucman** [comment]
nice... well I think something among those lines need to be mentionned in the doc... it's really not obvious how to do that or what other tool would bind what where to use it properly...

File: src/core/socket.c [hunk 1]
```diff
@@ -1473,6 +1473,25 @@ static int socket_address_listen_do(
                 log_unit_error_errno(u, error, fmt, strna(_t));  \
         })
 
+static int fork_needed(const SocketAddress *address, const ExecContext *context) {
+        int r;
+
+        assert(address);
+        assert(context);
+
+        /* Check if we need to do the cgroup or netns stuff. If not we can do things much simpler. */
```
Comment (keszybz) [inline src/core/socket.c:1482]: "do ... much more simply" (adverb) or "make things much simpler" (adjective).

**Pull request merged**